WordPress/src/main/java/org/wordpress/android/ui/posts/PostListFragment.kt

```diff
@@ -389,14 +372,94 @@ class PostListFragment : Fragment(),
+            return
+        }
+
+        // remove post from the list and add it to the list of trashed posts
+        val postIdPair = Pair(post.id, post.remotePostId)
+        trashedPostIds.add(postIdPair)
+        refreshListManagerFromStore(listDescriptor, shouldRefreshFirstPageAfterLoading = false)
+
+        val undoListener = OnClickListener {
+            // user undid the trash, so un-hide the post and remove it from the list of trashed posts
+            trashedPostIds.remove(postIdPair)
+            refreshListManagerFromStore(listDescriptor, shouldRefreshFirstPageAfterLoading = false)
+        }
+
+        // different undo text if this is a local draft since it will be deleted rather than trashed
+        val text = if (post.isLocalDraft) getString(R.string.post_deleted) else getString(R.string.post_trashed)
+        val snackbar = Snackbar.make(
+                nonNullActivity.findViewById(R.id.root_view),
+                text,
+                AccessibilityUtils.getSnackbarDuration(nonNullActivity)
+        ).setAction(R.string.undo, undoListener)
+        // wait for the undo snackbar to disappear before actually deleting the post
+        snackbar.addCallback(object : Snackbar.Callback() {
+            override fun onDismissed(snackbar: Snackbar?, event: Int) {
+                super.onDismissed(snackbar, event)
+
+                // if the post no longer exists in the list of trashed posts it's because the
+                // user undid the trash, so don't perform the deletion
+                if (!trashedPostIds.contains(postIdPair)) {
+                    return
+                }
+
+                // remove from the list of trashed posts in case onDismissed is called multiple
+                // times - this way the above check prevents us making the call to delete it twice
+                // https://code.google.com/p/android/issues/detail?id=190529
+                trashedPostIds.remove(postIdPair)
+
+                // here cancel all media uploads related to this Post
+                UploadService.cancelQueuedPostUploadAndRelatedMedia(WordPress.getContext(), post)
+
+                if (post.isLocalDraft) {
+                    dispatcher.dispatch(PostActionBuilder.newRemovePostAction(post))
+
+                    // delete the pending draft notification if available
+                    shouldCancelPendingDraftNotification = false
+                    val pushId = PendingDraftsNotificationsUtils.makePendingDraftNotificationId(post.id)
+                    NativeNotificationsUtils.dismissNotification(pushId, WordPress.getContext())
+                } else {
+                    dispatcher.dispatch(PostActionBuilder.newDeletePostAction(RemotePostPayload(post, site)))
+                }
+            }
+        })
+
+        postIdForPostToBeDeleted = post.id
+        shouldCancelPendingDraftNotification = true
+        snackbar.show()
+    }
+
+    private fun showPublishConfirmationDialog(post: PostModel) {
```

Comment:
The only change is the `isAdded` check, here is the version from base branch for easy comparison: https://github.com/wordpress-mobile/WordPress-Android/blob/feature/list-store-integration-for-post-list-master/WordPress/src/main/java/org/wordpress/android/ui/posts/PostListFragment.kt#L494-L511